[
    {
        "type": "error",
        "module": "src.api.ingest",
        "obj": "record_job_event",
        "line": 406,
        "column": 49,
        "endLine": 406,
        "endColumn": 71,
        "path": "src/api/ingest.py",
        "symbol": "no-member",
        "message": "Module 'secrets' has no 'compare_digest' member",
        "message-id": "E1101"
    },
    {
        "type": "refactor",
        "module": "docai_batch_helper",
        "obj": "batch_process_documents_gcs",
        "line": 116,
        "column": 0,
        "endLine": 116,
        "endColumn": 31,
        "path": "src/services/docai_batch_helper.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (6/5)",
        "message-id": "R0913"
    },
    {
        "type": "convention",
        "module": "docai_helper",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "src/services/docai_helper.py",
        "symbol": "too-many-lines",
        "message": "Too many lines in module (1355/1000)",
        "message-id": "C0302"
    },
    {
        "type": "warning",
        "module": "docai_helper",
        "obj": "_extract_document_dict",
        "line": 169,
        "column": 16,
        "endLine": 169,
        "endColumn": 23,
        "path": "src/services/docai_helper.py",
        "symbol": "protected-access",
        "message": "Access to a protected member _pb of a client class",
        "message-id": "W0212"
    },
    {
        "type": "refactor",
        "module": "docai_helper",
        "obj": "_is_retryable_error",
        "line": 256,
        "column": 0,
        "endLine": 256,
        "endColumn": 23,
        "path": "src/services/docai_helper.py",
        "symbol": "too-many-return-statements",
        "message": "Too many return statements (7/6)",
        "message-id": "R0911"
    },
    {
        "type": "refactor",
        "module": "docai_helper",
        "obj": "_process_chunk_with_docai",
        "line": 309,
        "column": 0,
        "endLine": 309,
        "endColumn": 29,
        "path": "src/services/docai_helper.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (6/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "docai_helper",
        "obj": "OCRService._process_large_pdf",
        "line": 641,
        "column": 4,
        "endLine": 641,
        "endColumn": 26,
        "path": "src/services/docai_helper.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (7/5)",
        "message-id": "R0913"
    },
    {
        "type": "error",
        "module": "metrics",
        "obj": "",
        "line": 13,
        "column": 0,
        "endLine": 13,
        "endColumn": 37,
        "path": "src/services/metrics.py",
        "symbol": "relative-beyond-top-level",
        "message": "Attempted relative import beyond top-level package",
        "message-id": "E0402"
    },
    {
        "type": "warning",
        "module": "metrics",
        "obj": "PrometheusMetrics.instrument_app",
        "line": 74,
        "column": 8,
        "endLine": 74,
        "endColumn": 42,
        "path": "src/services/metrics.py",
        "symbol": "protected-access",
        "message": "Access to a protected member _prometheus_instrumented of a client class",
        "message-id": "W0212"
    },
    {
        "type": "error",
        "module": "pdf_writer",
        "obj": "ReportLabBackend.build",
        "line": 41,
        "column": 12,
        "endLine": 41,
        "endColumn": 54,
        "path": "src/services/pdf_writer.py",
        "symbol": "import-error",
        "message": "Unable to import 'reportlab.lib.pagesizes'",
        "message-id": "E0401"
    },
    {
        "type": "error",
        "module": "pdf_writer",
        "obj": "ReportLabBackend.build",
        "line": 42,
        "column": 12,
        "endLine": 42,
        "endColumn": 47,
        "path": "src/services/pdf_writer.py",
        "symbol": "import-error",
        "message": "Unable to import 'reportlab.pdfgen'",
        "message-id": "E0401"
    },
    {
        "type": "error",
        "module": "pdf_writer",
        "obj": "PDFWriter.build",
        "line": 253,
        "column": 12,
        "endLine": 253,
        "endColumn": 65,
        "path": "src/services/pdf_writer.py",
        "symbol": "assignment-from-no-return",
        "message": "Assigning result of a function call, where the function has no return",
        "message-id": "E1111"
    },
    {
        "type": "warning",
        "module": "pipeline",
        "obj": "PipelineJob.<lambda>",
        "line": 114,
        "column": 46,
        "endLine": 114,
        "endColumn": 65,
        "path": "src/services/pipeline.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "warning",
        "module": "pipeline",
        "obj": "PipelineJob.<lambda>",
        "line": 115,
        "column": 46,
        "endLine": 115,
        "endColumn": 65,
        "path": "src/services/pipeline.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "refactor",
        "module": "pipeline",
        "obj": "PipelineStateStore.mark_status",
        "line": 150,
        "column": 4,
        "endLine": 150,
        "endColumn": 19,
        "path": "src/services/pipeline.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (7/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "pipeline",
        "obj": "InMemoryStateStore.mark_status",
        "line": 291,
        "column": 4,
        "endLine": 291,
        "endColumn": 19,
        "path": "src/services/pipeline.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (7/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "pipeline",
        "obj": "GCSStateStore.mark_status",
        "line": 494,
        "column": 4,
        "endLine": 494,
        "endColumn": 19,
        "path": "src/services/pipeline.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (7/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "pipeline",
        "obj": "NoopWorkflowLauncher.launch",
        "line": 672,
        "column": 4,
        "endLine": 672,
        "endColumn": 14,
        "path": "src/services/pipeline.py",
        "symbol": "useless-return",
        "message": "Useless return at end of function or method",
        "message-id": "R1711"
    },
    {
        "type": "convention",
        "module": "summariser_refactored",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "src/services/summariser_refactored.py",
        "symbol": "too-many-lines",
        "message": "Too many lines in module (1694/1000)",
        "message-id": "C0302"
    },
    {
        "type": "error",
        "module": "summariser_refactored",
        "obj": "OpenAIResponsesBackend.summarise_chunk",
        "line": 158,
        "column": 23,
        "endLine": 167,
        "endColumn": 13,
        "path": "src/services/summariser_refactored.py",
        "symbol": "unexpected-keyword-arg",
        "message": "Unexpected keyword argument 'response_format' in method call",
        "message-id": "E1123"
    },
    {
        "type": "warning",
        "module": "summariser_refactored",
        "obj": "OpenAIResponsesBackend.summarise_chunk",
        "line": 138,
        "column": 8,
        "endLine": 138,
        "endColumn": 27,
        "path": "src/services/summariser_refactored.py",
        "symbol": "attribute-defined-outside-init",
        "message": "Attribute '_fallback_used' defined outside __init__",
        "message-id": "W0201"
    },
    {
        "type": "warning",
        "module": "summariser_refactored",
        "obj": "OpenAIResponsesBackend.summarise_chunk",
        "line": 178,
        "column": 12,
        "endLine": 178,
        "endColumn": 31,
        "path": "src/services/summariser_refactored.py",
        "symbol": "attribute-defined-outside-init",
        "message": "Attribute '_fallback_used' defined outside __init__",
        "message-id": "W0201"
    },
    {
        "type": "error",
        "module": "summariser_refactored",
        "obj": "RefactoredSummariser.summarise",
        "line": 680,
        "column": 12,
        "endLine": 685,
        "endColumn": 13,
        "path": "src/services/summariser_refactored.py",
        "symbol": "assignment-from-no-return",
        "message": "Assigning result of a function call, where the function has no return",
        "message-id": "E1111"
    },
    {
        "type": "refactor",
        "module": "summariser_refactored",
        "obj": "RefactoredSummariser._is_noise_line",
        "line": 826,
        "column": 4,
        "endLine": 826,
        "endColumn": 22,
        "path": "src/services/summariser_refactored.py",
        "symbol": "too-many-return-statements",
        "message": "Too many return statements (11/6)",
        "message-id": "R0911"
    },
    {
        "type": "refactor",
        "module": "summariser_refactored",
        "obj": "RefactoredSummariser._dedupe_ordered",
        "line": 962,
        "column": 4,
        "endLine": 962,
        "endColumn": 23,
        "path": "src/services/summariser_refactored.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (6/5)",
        "message-id": "R0913"
    },
    {
        "type": "convention",
        "module": "summariser_refactored",
        "obj": "_cli",
        "line": 1625,
        "column": 35,
        "endLine": 1625,
        "endColumn": 49,
        "path": "src/services/summariser_refactored.py",
        "symbol": "consider-iterating-dictionary",
        "message": "Consider iterating the dictionary directly instead of calling .keys()",
        "message-id": "C0201"
    },
    {
        "type": "refactor",
        "module": "supervisor",
        "obj": "CommonSenseSupervisor.__init__",
        "line": 95,
        "column": 4,
        "endLine": 95,
        "endColumn": 16,
        "path": "src/services/supervisor.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (7/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "supervisor",
        "obj": "CommonSenseSupervisor.validate",
        "line": 143,
        "column": 4,
        "endLine": 143,
        "endColumn": 16,
        "path": "src/services/supervisor.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (6/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "supervisor",
        "obj": "CommonSenseSupervisor.retry_and_merge",
        "line": 332,
        "column": 4,
        "endLine": 332,
        "endColumn": 23,
        "path": "src/services/supervisor.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (6/5)",
        "message-id": "R0913"
    },
    {
        "type": "refactor",
        "module": "docai_request_builder",
        "obj": "build_docai_request",
        "line": 48,
        "column": 0,
        "endLine": 48,
        "endColumn": 23,
        "path": "src/utils/docai_request_builder.py",
        "symbol": "too-many-arguments",
        "message": "Too many arguments (7/5)",
        "message-id": "R0913"
    },
    {
        "type": "warning",
        "module": "secrets",
        "obj": "",
        "line": 10,
        "column": 4,
        "endLine": 10,
        "endColumn": 50,
        "path": "src/utils/secrets.py",
        "symbol": "unused-import",
        "message": "Unused exceptions imported from google.api_core as gexc",
        "message-id": "W0611"
    }
]
